src/config_tools/config_tools.cc

```diff
@@ -235,57 +242,83 @@ void ValidateIOConfig(TritonJson::Value &io_object, const IOConfig &io_config) {
 }
 
 
-template <bool input>
-void AutofillIOsConfig(TritonJson::Value &ios, const std::vector<IOConfig> &io_configs,
-                       TritonJson::Value &new_ios) {
-  TRITON_CALL(ios.AssertType(common::TritonJson::ValueType::ARRAY));
-  TRITON_CALL(new_ios.AssertType(common::TritonJson::ValueType::ARRAY));
-
-  std::vector<TritonJson::Value> new_io_objs(io_configs.size());
-  auto end_ind = ios.ArraySize();
-  for (const auto &io_config: io_configs) {
+void ValidateAgainstTooManyInputs(TritonJson::Value &ins, const std::vector<IOConfig> &in_configs) {
+   for (size_t i = 0; i < ins.ArraySize(); ++i) {
     TritonJson::Value io_object(TritonJson::ValueType::OBJECT);
-    auto ind = FindObjectByName(ios, io_config.name, &io_object);
-    size_t io_index;
+    ins.IndexAsObject(i, &io_object);
+    std::string name;
+    if (io_object.MemberAsString("name", &name) != TRITONJSON_STATUSSUCCESS) {
+      throw TritonError::InvalidArg(make_string("Missing name in IO config at position ", i));
+    }
+
+    bool in_present = std::any_of(in_configs.begin(), in_configs.end(),
+                                  [&](const auto &ioc) { return ioc.name == name; });
+    if (!in_present) {
+      throw TritonError::InvalidArg(make_string("Configuration file contains config for ", name,
+                                                " but such input is not present in the pipeline."));
```

Comment:
I started to wonder, is it necessary to restrict number of inputs this way? I mean, if there are any inputs in the model config, that are not present in the DALI pipeline, we can proceed, at least for now. Eventually, when user feeds the request with the input not present in the DALI pipeline, we can throw error at that stage. This way model config, especially during development, does not have to be modified frequently. What do you think?

Reply:
There is no way for a config with too many inputs to be useful - the request you send is validated against that config, so it has to contain all inputs named in the config file and there's no way to feed those inputs to the pipeline, so we would have to just error out.

Reply:
Ok, I see that in fact we could just drop the inputs that or not used by the pipelines. 
Still, I think we should guard against such situation because in normal scenario (not developing or debugging) sending inputs that are not used is wasteful and imposes performance overhead.